File: backend-php/models/SosAlert.php
<?php
/**
 * Entité SosAlert : alerte SOS avec géolocalisation.
 * Équivalent PHP de com.ihecride.models.SosAlert.
 */

class SosAlert implements JsonSerializable {

    public ?int    $id = null;
    public ?int    $userId = null;
    public ?float  $latitude = null;
    public ?float  $longitude = null;
    public ?string $timestamp = null;
    public ?string $statut = null;        // ENVOYE, TRAITE

    public function jsonSerialize(): array {
        return [
            'id'         => $this->id,
            'userId'     => $this->userId,
            'latitude'   => $this->latitude,
            'longitude'  => $this->longitude,
            'timestamp'  => $this->timestamp,
            'statut'     => $this->statut,
        ];
    }
}

File: backend/src/main/java/com/ihecride/models/SosAlert.java
package com.ihecride.models;

import java.math.BigDecimal;
import java.sql.Timestamp;

public class SosAlert {
    private int id;
    private int userId;
    private BigDecimal latitude;
    private BigDecimal longitude;
    private Timestamp timestamp;
    private String statut;  // ENVOYE, TRAITE

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }
    public BigDecimal getLatitude() { return latitude; }
    public void setLatitude(BigDecimal latitude) { this.latitude = latitude; }
    public BigDecimal getLongitude() { return longitude; }
    public void setLongitude(BigDecimal longitude) { this.longitude = longitude; }
    public Timestamp getTimestamp() { return timestamp; }
    public void setTimestamp(Timestamp timestamp) { this.timestamp = timestamp; }
    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }
}
